zerver/lib/outgoing_webhook.py

```diff
@@ -161,14 +159,20 @@ def fail_with_message(event, failure_message):
 
 def request_retry(event, failure_message):
     # type: (Dict[str, Any], Text) -> None
-    event['failed_tries'] += 1
-    if event['failed_tries'] > MAX_REQUEST_RETRIES:
+    def failure_processor(event):
+        # type: (Dict[str, Any]) -> None
+        """
+        The name of the argument is 'event' on purpose. This argument will hide
+        the 'event' argument of the request_retry function. Keeping the same name
+        results in a smaller diff.
+        """
         bot_user = get_user_profile_by_id(event['user_profile_id'])
+        nonlocal failure_message
```

Comment:
I think this is the wrong fix; we should just not mutate `failure_message` inside this function.